Lib/test/test_readline.py

```diff
@@ -210,13 +212,49 @@ def display(substitution, matches, longest_match_length):
         self.assertIn(b"result " + expected + b"\r\n", output)
         self.assertIn(b"history " + expected + b"\r\n", output)
 
+    def test_history_size(self):
+        history_size = 10
+        temp_dir = tempfile.mkdtemp()
+        try:
+            inputrc = os.path.join(temp_dir, "inputrc")
+            with io.open(inputrc, "wb") as f:
+                f.write(b"set history-size %d\n" % history_size)
+            env = os.environ.copy()
```

Comment:
I think it would be safer and clearer to make the copy with _env = dict(os.environ)_. _Os.environ.copy_ is undocumented.